nmostesting/suites/IS1101Test.py

```diff
@@ -1715,3 +1716,419 @@ def test_04_02_02(self, test):
             if len(self.caps["constraint_sets"]) == 0:
                 return test.WARNING("The receiver does not support BCP-004-01.")
         return test.PASS()
+
+    def test_04_03(self, test):
+        """
+        Verify receivers supporting outputs
+        """
+        if len(self.receivers) == 0:
+            return test.UNCLEAR("No IS-11 receivers")
+        for receiver_id in self.receivers:
+            _, response = TestHelper.do_request('GET', self.compat_url + "receivers/")
+            if response.status_code != 200:
+                return test.FAIL("The receiver's streamcompatibility request has failed: {}"
+                                 .format(response.json()))
+            self.receivers_outputs = response.json()
+
+            """
+            The test requires streaming from a Sender in order to
+            verify the state of the Receiver and the associated outputs.
+            """
+            """
+            if IS11_RefecenceSender is not configured fail 4.3 with log indicating
+            that IS11_RefecenceSender is not configured
+            """
+            if len(CONFIG.IS11_REFERENCE_SENDER) == 0:
+                return test.FAIL("IS11_RefecenceSender is not configured")
+            if len(self.receivers_outputs) == 0:
+                return test.UNCLEAR("No IS-11 receivers support outputs")
+            for receiver_id in self.receivers_outputs:
+                _, response = TestHelper.do_request('GET', self.node_url + "receivers/" + receiver_id)
+                if response.status_code != 200:
+                    return test.FAIL("The Node API request for receiver {} has failed: {}"
+                                     .format(receiver_id, response.json()))
+                receiver_format = response.json()["format"]
+                receiver_transport_file = response.json()["transport"]
+                if (receiver_format == "urn:x-nmos:format:video"):
+                    """
+                    using $IS11_RefecenceSender GET sender_id from N?/senders/*/ where
+                    .format is receiver.format and .transport is receiver.transport
+                    """
+                    url = self.node_url.replace(PATTERN.search(self.node_url)[0], CONFIG.IS11_REFERENCE_SENDER)
```

Comment:
This seems unacceptable way to create URL.

Reply:
`node_url` might use a host name rather than an IPv4 address. It might have a port specified.
The rest of the testing tool codebase uses [`urlparse`](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlparse).

But as mentioned above, in any case, one authority (`hostname:port`) should not be used for three different APIs, as in general these may not be provided on the same URL/port.